hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -1,57 +1,542 @@
 // Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
-{ name: "SPI_HOST",
+{ name: "spi_host",
   clock_primary: "clk_i",
+  other_clock_list: [ "clk_core_i" ],
+  reset_primary: "rst_ni",
+  other_reset_list: [ "rst_core_ni" ],
   bus_device: "tlul",
-  available_input_list: [
+  bus_host: "none",
+  regwidth: "32",
+  param_list: [
+    { name: "ByteOrder",
+      desc: '''Byte order to use when transmitting or receiving data.  If ByteOrder = 0,
+               the IP uses a Big-Endian ordering for the bytes in TXDATA and RXDATA.
+               The most significant byte (MSB) of TXDATA is transmitted first, and
+               received data is placed in the MSB location of RXDATA.  If ByteOrder = 1,
+               a Little-Endian ordering is used for these registers, and the LSB of each
+               gets priority for receiving and transmitting data.'''
+      type: "int",
+      default: "0"
+    },
+    { name: "MaxCS",
+      desc: "The number of active-low chip select (cs_n) lines to create.",
+      type: "int",
+      default: "1"
+    },
   ],
   available_output_list: [
-    { name: "sck", desc: "SPI Clock" },
-    { name: "csb", desc: "Chip Enable#" },
+    { name:  "sck"
+      desc:  "SPI Clock"
+    },
+    { name:  "csb"
+      desc:  '''Chip Select# (One hot, active low).  The size of this port should match MaxCS.'''
+      width: "1"
```

Comment:
@msfschaffner 
@martin-lueker 
@eunchan 
@rswarbrick 

we should probably improve the module merging to allow us to use parameters when defining the width. 
it feels like there are a few places that could benefit from this.  Rupert in the past has also suggested we switch to something like pyparsing to better support arithmetic on parameters.

Reply:
I agree.  Looking at the rest of the IP's, it's not even clear to me how to modify parameters at top level.

There is also the issue that it seems that regtool doesn't like parameterized pin counts.

Reply:
It won't help with the parameterised pin counts, but the regtool work I've been doing recently tidies up parameter expansion quite a bit. I'll push up the parameter code as the next step once #5283 is merged.

Reply:
thanks @rswarbrick ! I'm hoping we can adopt the same way this is handled and apply it to the merge step for pinmux.